cpp/mrc/src/internal/control_plane/client.cpp

```diff
@@ -260,4 +277,37 @@ void Client::request_update()
     // }
 }
 
+AsyncEventStatus Client::write_event(protos::Event event, bool await_response)
+{
+    if (event.tag() != 0)
+    {
+        LOG(WARNING) << "event tag is set but this field should exclusively be used by the control plane client. "
+                        "Clearing to avoid confusion";
+        event.clear_tag();
+    }
```

Comment:
Suggestion: Would be good to add a test for this code path.

Reply:
This method (`Client::write_event`) is private, and none of the public client methods provide a way of indirectly setting the `event.tag` value.

I could change the method to protected and then subclass it in a test.

Reply:
>I could change the method to protected and then subclass it in a test.

On second thought, this isnt worth the effort.